### PR TITLE
Truncate output that we log from commands to 10 lines

### PR DESCRIPTION
**Pull request by ericholscher**
Refs https://github.com/readthedocs/readthedocs.org/issues/9105

### CODE REVIEW

File: readthedocs/doc_builder/environments.py [hunk 1]
```diff
@@ -457,7 +457,8 @@ def run_command_class(
             if warn_only:
                 msg = 'Command {cmd} failed'.format(cmd=build_cmd.get_command())
                 if build_cmd.output:
-                    msg += ':\n{out}'.format(out=build_cmd.output)
+                    truncated_output = "\n".join(build_cmd.output.split("\n")[:10])
+                    msg += ":\n{out}".format(out=truncated_output)
                 log.warning(
                     msg,
```
Comment (stsewd) [inline readthedocs/doc_builder/environments.py:463]: Probably we should set the msg to "Command failed" and log the command and the output as kwargs of `log.warning`, this is so we can filter all these easily

File: readthedocs/doc_builder/environments.py [hunk 2]
```diff
@@ -457,7 +457,8 @@ def run_command_class(
             if warn_only:
                 msg = 'Command {cmd} failed'.format(cmd=build_cmd.get_command())
                 if build_cmd.output:
-                    msg += ':\n{out}'.format(out=build_cmd.output)
+                    truncated_output = "\n".join(build_cmd.output.split("\n")[:10])
```
Comment (humitos) [inline readthedocs/doc_builder/environments.py:460]: What about logging the 10 first and the 10 last lines?

```
8
9
10

....

98
99
100
```

File: readthedocs/doc_builder/environments.py (hunk 1 above)
Comment (humitos) [reply readthedocs/doc_builder/environments.py:463]: I like this idea too.

**Review by humitos — APPROVED** [review]
Looks good! There are some feedback that could be applied as well.

**Review by humitos — APPROVED** [review]
🤝

**Comment by ericholscher** [comment]
Failing test is unrelated (jupyterbook docs change)

**Pull request merged**